File: classes/SpaceObject.php
<?php
class SpaceObject
{
    private static $instances = [];
    public $name;
    public $orbit_to;
    public $orbit_from = [];

    public function __construct($name)
    {
        $this->name = $name;
    }

    public function setOrbitTo(SpaceObject $orbit_to)
    {
        $this->orbit_to = $orbit_to;
    }

    public function setOrbitFrom(SpaceObject $orbit_from)
    {
        $this->orbit_from[] = $orbit_from;
    }

    public static function getInstance($name)
    {
        if (!array_key_exists($name, self::$instances)) {
            self::$instances[$name] = new SpaceObject($name);
        }
        return self::$instances[$name];
    }
}

File: machines/space_object.py
from typing import Dict, List


class SpaceObject:
    __instacnces: Dict = {}
    name: str
    orbit_to: "SpaceObject"
    orbit_from: List["SpaceObject"]

    def __init__(self, name):
        self.name = name
        self.orbit_to = None
        self.orbit_from = []

    def set_orbit_to(self, orbit_to: "SpaceObject"):
        self.orbit_to = orbit_to

    def set_orbit_from(self, orbit_from: "SpaceObject"):
        self.orbit_from += [orbit_from]

    @classmethod
    def get_instance(cls, name):
        if name not in cls.__instacnces:
            cls.__instacnces[name] = SpaceObject(name)
        return cls.__instacnces[name]
